Documentation/project-docs/linux-performance-tracing.md

```diff
@@ -0,0 +1,114 @@
+Performance Tracing on Linux
+============================
+
+When a performance problem is encountered on Linux, these instructions can be used to gather detailed information about what was happening on the machine at the time of the performance problem.
+
+#Required Tools#
+- **Perfcollect**: Bash script that automates data collection.
+	- Available at <http://aks.ms/perfcollect>.
+- **PerfView**: Windows-based performance tool that can also analyze trace files collected with Perfcollect.
+	- Available at <http://aks.ms/perfview>.
+
+#Preparing Your Machine#
+Follow these steps to prepare your machine to collect a performance trace.
+
+1. Download Perfcollect.
+
+	> ```bash
+	> curl -OL http://aks.ms/perfcollect
+	> ```
+
+2. Make the script executable.
+
+	> ```bash
+	> chmod +x perfcollect
+	> ```
+
+3. Install tracing prerequisites - these are the actual tracing libraries.  For details on prerequisites, see [below](#prerequisites).
+
+	> ```bash
+	> sudo ./perfcollect install
+	> ```
+
+#Collecting a Trace#
+1. Have two shell windows available - one for controlling tracing, referred to as **[Trace]** and one for running the application, referred to as **[App]**.
+2. **[App]** Setup the application shell - this enables tracing configuration inside of CoreCLR.
+
+	> ```bash 
+	> export COMPlus_PerfMapEnabled=1
+	> export COMPlus_EnableEventLog=1
+	> ```
+
+3. **[Trace]** Start collection. 
+
+	> ```bash
+	> sudo ./perfcollect collect sampleTrace
+	> Collection started.  Press CTRL+C to stop.
```

Comment:
Most people reading this will probably understand, but it'd be good to differentiate what the user needs to type to stdin vs what the tool is going to spit out to stdout.  Folks shouldn't be typing in that second line, in other words.